### PR TITLE
Allow 1 or more satellite resources languages to filter assets

### LINKED ISSUE #17184: SatelliteResourceLanguages doesn't filter resources from WindowsDesktop sdk on build
When using a RuntimeIdentifier to build standalone with UseWpf, the build output contains many folders like cs/, de/ etc - about 200 .resources.dll files total.

### PR DESCRIPTION
**Pull request by joeloff**
Fixes #17184

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolveRuntimePackAssetsTask.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+using Xunit;
+
+namespace Microsoft.NET.Build.Tasks.UnitTests
+{
+    public class GivenAResolveRuntimePackAssetsTask
+    {
+        [Fact]
+        public void ItFiltersSatelliteResources()
+        {
+            string projectDirectory = Path.Combine(Path.GetTempPath(), "runtimePackSatelliteAssets");
```
Comment (sfoslund) [inline src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolveRuntimePackAssetsTask.cs:18]: nit: can you use _testAssetsManager.CreateTestDirectory() instead? That way you won't have to delete it at the end.

**Review by sfoslund — APPROVED** [review]
LGTM, although isn't 5.0.3xx in M2 mode right now?

**Comment by joeloff** [comment]
@marcpopMSFT should I retarget to something else?

**Review by dsplaisted — COMMENTED** [review]
Note that there are existing tests covering `SatelliteResourceLanguages` for both Build and Publish under classes named `GivenThatWeWantToFilterSatelliteAssemblies`.

**Comment by marcpopMSFT** [comment]
@joeloff I'd retarget to main.  5.0.3xx is in QB mode now and we're being asked to scope 5.0.4xx as well.

**Pull request closed without merging**